packages/firestore/src/model/mutation.ts

```diff
@@ -385,23 +429,19 @@ export function mutationEquals(left: Mutation, right: Mutation): boolean {
   }
 
   if (left.type === MutationType.Set) {
-    return (left as SetMutation).value.isEqual((right as SetMutation).value);
+    return (
+      (left as SetMutation).value.isEqual((right as SetMutation).value) &&
+      fieldTransformsAreEqual(left.fieldTransforms, right.fieldTransforms)
```

Comment:
This check can now be extracted and performed before going into any of the if statements.

Reply:
Extracted.